utils/data.py

```
@@ -0,0 +1,76 @@
+    """
+    리스트 데이터를 JSON 파일로 저장.
+    디렉토리가 없으면 자동으로 생성함.
+    성공 시 True, 실패 시 False 반환.
+    """
+    file_path = DATA_DIR / filename
+
+    try:
+        file_path.parent.mkdir(parents=True, exist_ok=True)
+        with open(file_path, "w", encoding="utf-8") as f:
+            json.dump(data, f, ensure_ascii=False, indent=2)
+        return True
```

Ruff (0.14.13)

Consider moving this statement to an `else` block

(TRY300)